crates/stackable-telemetry/src/tracing/settings/mod.rs

```diff
@@ -0,0 +1,166 @@
+//! Subscriber settings.
+
+use tracing::level_filters::LevelFilter;
+
+pub mod console_log;
+pub use console_log::*;
+
+pub mod otlp_log;
+pub use otlp_log::*;
+
+pub mod otlp_trace;
+pub use otlp_trace::*;
+
+/// Simplifies access common settings from subscriber specific settings.
+pub trait CommonSettings {
+    /// Access to the [`Settings::environment_variable`] field.
+    fn environment_variable(&self) -> &'static str;
+
+    /// Access to the [`Settings::default_level`] field.
+    fn default_level(&self) -> LevelFilter;
```

Comment:
This could be isolated from each impl by defining a

```rust
fn common(&self) -> &Settings;
```

and then providing default impls for everything else that delegates to it:

```suggestion
    /// Access to the [`Settings::environment_variable`] field.
    fn environment_variable(&self) -> &'static str {
      self.common().environment_variable
    }

    /// Access to the [`Settings::default_level`] field.
    fn default_level(&self) -> LevelFilter {
      self.common().default_level
    }
```

Reply:
Agreed, great idea!

Reply:
Another alternative would be to have each variant provide a `Deref<Target = Settings>`.

Reply:
Oh yeah, also a nice idea. I will first try out the `common()` approach and then I can look into the `Deref` solution.

Reply:
First iteration in 5de6ed4

Reply:
Second iteration in 8f59b15